packages/starknet_provider/lib/src/model/get_transaction_status.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'get_transaction_status.freezed.dart';
+part 'get_transaction_status.g.dart';
+
+@freezed
+class TransactionStatus with _$TransactionStatus {
+  const factory TransactionStatus({
+    required String finalityStatus,
```

Comment:
According to [JSON RPC SPEC](https://github.com/starkware-libs/starknet-specs/blob/76bdde23c7dae370a3340e40f7ca2ef2520e75b9/api/starknet_api_openrpc.json#L225-L264), [finalityStatus](https://github.com/starkware-libs/starknet-specs/blob/76bdde23c7dae370a3340e40f7ca2ef2520e75b9/api/starknet_api_openrpc.json#L3200-L3209) should be an enum